src/module-elasticsuite-virtual-category/Model/Rule.php

```diff
@@ -306,9 +316,14 @@ private function buildCategorySearchQuery($category, $excludedCategories = []):
         if (!in_array($category->getId(), $excludedCategories)) {
             $excludedCategories[] = $category->getId();
 
-            if ((bool) $category->getIsVirtualCategory() && $category->getIsActive()) {
+            if (!$category->getIsActive()
+                && $this->config->isForceZeroResultsForDisabledCategoriesEnabled($this->getStoreId())) {
+                return $this->getNoResultsQuery();
+            }
+
+            if ($category->getIsVirtualCategory()) {
```

Comment:
Be careful, you are creating a specific route that is not correctly covered.
You cannot remove the test on `$category->getIsActive()` on lines 324 and 326 since the test on 319 also relies on the new setting.
So if "isForceZero..." returns false, the category could be disabled on lines 324 and 326...

Reply:
I thought we might don't want to return "null" query anymore that can have a quite unexpected behavior ?
The opposite of forcing an empty result for empty disabled category should be to treat them as normal categories no ?